massgov/mayflower-patternlab
# DP-8453: Prevent listing header from being loaded on every pageview

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/dev/docs/change-log-instructions.md)


## Description

The resultHeading Handlebars template is being loaded on every pageview, even on pages it's not needed before.  This HTTP request is made synchronously, meaning that it blocks further page rendering until it's completed.  This PR:

* Prevents fetching of the template until the result listing header template is actually needed.
* Converts to using TwigJS to load and render the result-headings where it's needed.
* Removes the resultHeading Handlebars template.
* Converts to using TwigJS to load and render the pagination where it's needed.
* Removes the pagination Handlebars template.

## Related Issue / Ticket

- [JIRA issue](https://jira.state.ma.us/browse/DP-8453)

## Steps to Test


1. Visit any pattern that does not require the resultHeading template (eg: https://mayflower.digital.mass.gov/index.html?p=pages-ERROR-403) and verify that it does not cause an HTTP request for the resultHeading template.
2. Visit a location listings page (eg: https://mayflower.digital.mass.gov/index.html?p=pages-location-listing), submit the filter form and verify that it rerenders the result headings properly using javascript.  Also verify that it rerenders the pagination properly using javascript. 




## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.
<img width="804" alt="screen shot 2018-04-09 at 7 49 14 am" src="https://user-images.githubusercontent.com/654407/38498594-8f64391a-3bca-11e8-93ec-c88ad2a1c77c.png">
<img width="1653" alt="screen shot 2018-04-09 at 7 49 26 am" src="https://user-images.githubusercontent.com/654407/38498608-98a1f86e-3bca-11e8-8020-c2258089d9b2.png">


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


* 

#### @TODO


*

#### Today I learned...
